Sources/NetworkProtection/Settings/Extensions/UserDefaults+dnsSettings.swift

```diff
@@ -0,0 +1,85 @@
+//
+//  UserDefaults+dnsSettings.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Combine
+import Foundation
+
+extension UserDefaults {
+    final class StorableDNSSettings: NSObject, Codable {
+        let usesCustomDNS: Bool
+        let dnsServers: [String]
+
+        init(usesCustomDNS: Bool = false, dnsServers: [String] = []) {
+            self.usesCustomDNS = usesCustomDNS
+            self.dnsServers = dnsServers
+        }
```

Comment:
Controversial suggestion: what if we make this a failable initializer, which returns nil if `usesCustomDNS` is true but `dnsServers` doesn't meet our criteria? (i.e., the array is empty, the strings aren't valid IPs, etc.)

Reply:
@samsymons I'm doing the data sanitization in the setter; enforcing this in StorableDNSSettings feels a level too low, if that makes sense?

https://github.com/duckduckgo/BrowserServicesKit/blob/f63347234551c90fd3676dbf9f41d4e9b2a6c0a9/Sources/NetworkProtection/Settings/Extensions/UserDefaults%2BdnsSettings.swift#L61-L73

Reply:
Overall I'd say the current solution is fine, mainly what I'm getting at is that it's possible today to create an invalid  `StorableDNSSettings` object and having it done at the init level would make it impossible. But, given the usage here is so specific, I think we can go with it. :+1: